src/main/java/org/betonquest/betonquest/events/PickRandomEvent.java

```diff
@@ -70,53 +74,35 @@ public PickRandomEvent(final Instruction instruction) throws InstructionParseExc
 
     @Override
     protected Void execute(final Profile profile) throws QuestRuntimeException {
-        final List<RandomEvent> events = new ArrayList<>(this.events);
-        double total = 0;
-        // Calculate total amount of all "percentages" (so that it must not be 100)
-        for (final RandomEvent event : events) {
-            total += event.getChance().getDouble(profile);
+        final List<ResolvedRandomEvent> events = new ArrayList<>();
```

Comment:
Using an ArrayList here is very inefficient:
 * The only reading happens in order (via iterating over it) (ArrayList is optimal for random access)
 * The only write operation is a remove at a random position (ArrayList is optimal for removes at the start / end).
```suggestion
        final List<ResolvedRandomEvent> events = new LinkedList<>();
```